File: js/management-api-v2/cm_api_v2_patch_type.js
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
// Using ES6 syntax
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: '<YOUR_ENVIRONMENT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

const response = await client.modifyContentType()
  .byTypeId("0be13600-e57c-577d-8108-c8d860330985")
  // .byTypeCodename("my_article")
  // .byTypeExternalId("my-article-id")
  .withData(
    [
      {
        op: "replace",
        path: "/name",
        value: "A new type name"
      },
      {
        op: "replace",
        path: "/elements/codename:my_text_element/guidelines",
        value: "Here you can tell users how to fill in the element."
      },
      {
        op: "addInto",
        path: "/elements",
        value: {
          name: "My title",
          type: "text",
          guidelines: "Title of the article in plain text.",
          external_id: "my-title-id"
        }
      },
      {
        op: "remove",
        path: "/elements/id:0b2015d0-16ae-414a-85f9-7e1a4b3a3eae"
      },
      {
        op: "remove",
        path: "/elements/external_id:my-multiple-choice-id/options/codename:my_option"
      },
      {
        op: "remove",
        path: "/elements/id:e94fab1f-c2c1-4f4a-b36c-3f7a0808d2b8/allowed_blocks/text"
      }
    ]
  )
  .toPromise();

File: net/management-api-v2/cm_api_v2_patch_type.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;
using Kontent.Ai.Management.Configuration;
using Kontent.Ai.Management.Models.Shared;
using Kontent.Ai.Management.Models.Types.Elements;
using Kontent.Ai.Management.Models.Types.Elements.DefaultValues;
using Kontent.Ai.Management.Models.Types.Patch;

// Or register it through DI with services.AddManagementClient()
using var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var identifier = Reference.ById(Guid.Parse("0be13600-e57c-577d-8108-c8d860330985"));
// var identifier = Reference.ByCodename("my_article");
// var identifier = Reference.ByExternalId("my-article-id");

var response = (await client.ModifyContentTypeAsync(identifier,
[
    ContentTypePatch.ReplaceName("A new type name"),
    ContentTypePatch.ReplaceGuidelines(
        Reference.ByCodename("my_text_element"),
        "Here you can tell users how to fill in the element."),
    ContentTypePatch.ReplaceDefault(
        Reference.ByCodename("my_text_element"),
        new TextElementDefaultValueModel("This is a default value of the text element.")),
    ContentTypePatch.AddElement(new TextElementMetadataModel
    {
        Name = "My title",
        Guidelines = "Title of the article in plain text.",
        ExternalId = "my-title-id",
    }),
    ContentTypePatch.RemoveElement(Reference.ById(Guid.Parse("0b2015d0-16ae-414a-85f9-7e1a4b3a3eae"))),
    ContentTypePatch.MoveElementAfter(
        Reference.ByCodename("my_text_element"),
        Reference.ByExternalId("my-title-id")),
    ContentTypePatch.MoveOptionBefore(
        Reference.ByExternalId("my-multiple-choice-id"),
        Reference.ById(Guid.Parse("d66ffa49-86ff-eeaa-c33b-e5d9eefe8b81")),
        Reference.ById(Guid.Parse("523e6231-8d80-a158-3601-dffde4e64a78")))
])).EnsureSuccess();
